### PR TITLE
Add test for hipStreamAttachMemAsync

### PR DESCRIPTION
**Pull request by jatinx**

### CODE REVIEW

**Review by gandryey — DISMISSED** [review]
hipStreamAttachMemAsync is a nop under the current HMM implementation. All allocations have global access all time.

File: tests/catch/unit/stream/hipStreamAttachMemAsync.cc [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  if (streamType == StreamAttachTestType::StreamPerThread) {
+    stream = hipStreamPerThread;
+  } else if (streamType == StreamAttachTestType::CreatedStream) {
+    HIP_CHECK(hipStreamCreate(&stream));
+    REQUIRE(stream != nullptr);
+  }
+
+  SECTION("Invalid Resource Handle") {
+    int definitelyNotAManagedVariable = 0;
+    HIP_CHECK_ERROR(hipStreamAttachMemAsync(stream, &definitelyNotAManagedVariable, sizeof(int),
```
Comment (mangupta) [inline tests/catch/unit/stream/hipStreamAttachMemAsync.cc]: dev_ptr needs to be type void* and not int*. Causes compilation failures. Similar issue is seen in the rest of the file as well.

**Review by mangupta — CHANGES_REQUESTED** [review]

File: tests/catch/unit/stream/hipStreamAttachMemAsync.cc (hunk 1 above)
Comment (jatinx) [reply tests/catch/unit/stream/hipStreamAttachMemAsync.cc]: done

**Review by chrispaquot — DISMISSED** [review]

File: tests/catch/unit/stream/CMakeLists.txt [hunk 2]
```diff
@@ -25,9 +25,13 @@ set(TEST_SRC
     hipStreamCreateWithFlags.cc
     hipStreamCreateWithPriority.cc
     hipAPIStreamDisable.cc
+    # hipStreamAttachMemAsync.cc   # Disabling it on nvidia due to issue in function definition of hipStreamAttachMemAsync
```
Comment (chrispaquot) [inline tests/catch/unit/stream/CMakeLists.txt:28]: @gandryey Any comment on this?

Comment (gandryey) [reply tests/catch/unit/stream/CMakeLists.txt:28]: hipStreamAttachMemAsync is a nop for AMD. I'm not entirely sure what should be done there. We always have global attachment and transfer data per a GPU page fault. It seems nvidia had some kind of SW management before and that call could optimize data transfers per stream.

**Review by gandryey — APPROVED** [review]

**Pull request merged**